iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -269,21 +281,62 @@ def _adjust_operator_bundle(manifests_path, metadata_path, organization=None, pi
             )
             continue
 
-        operator_csvs.append(operator_csv)
+        bundle_metadata['operator_csvs'].append(operator_csv)
 
         for pullspec in operator_csv.get_pullspecs():
-            found_pullspecs.add(pullspec)
+            bundle_metadata['found_pullspecs'].add(pullspec)
+    return bundle_metadata
 
-    conf = get_worker_config()
-    registry_replacements = (
-        conf['iib_organization_customizations']
-        .get(organization, {})
-        .get('registry_replacements', {})
-    )
 
+def _get_package_annotations(metadata_path):
+    """
+    Get valid annotations yaml of the bundle.
+
+    :param str metadata_path: the path to the bundle's metadata directory.
+    :raises IIBError: if the annotations.yaml has invalid entries.
+    :return: a dictionary of the bundle annotations.yaml file.
+    :rtype: dict
+    """
+    annotations_yaml_path = os.path.join(metadata_path, 'annotations.yaml')
+    if not os.path.exists(annotations_yaml_path):
+        raise IIBError('metadata/annotations.yaml does not exist in the bundle')
+
+    with open(annotations_yaml_path, 'r') as f:
+        try:
+            annotations_yaml = yaml.load(f)
+        except ruamel.yaml.YAMLError:
+            error = 'metadata/annotations.yaml is not valid YAML'
+            log.exception(error)
+            raise IIBError(error)
+
+    if not isinstance(annotations_yaml.get('annotations', {}), dict):
+        raise IIBError('The value of metadata/annotations.yaml must be a dictionary')
+
+    package_label = 'operators.operatorframework.io.bundle.package.v1'
+    package_annotation = annotations_yaml.get('annotations', {}).get(package_label)
+    if not package_annotation:
+        raise IIBError(f'{package_label} is not set in metadata/annotations.yaml')
+
+    if not isinstance(package_annotation, str):
+        raise IIBError(f'The value of {package_label} in metadata/annotations.yaml is not a string')
+
+    return annotations_yaml
+
+
+def _resolve_image_pull_specs(bundle_metadata, labels, pinned_by_iib, registry_replacements=None):
```

Comment:
Since you allow `None` for `registry_replacements`, could you please add this at the beginning of the function?
```python
if registry_replacements is None:
    registry_replacements = {}
```

Reply:
It's only being called from one place and an empty dict will always be passed. Maybe I'll not have this optional at all.